src/vm/database/marf.rs

```diff
@@ -104,35 +186,62 @@ impl <S> MarfedKV <S> where S: KeyValueStorage {
         self.side_store.rollback(&self.chain_tip);
         self.chain_tip = TrieFileStorage::block_sentinel();
     }
-    pub fn commit(&mut self) {
+    #[cfg(test)]
+    pub fn test_commit(&mut self) {
+        let bhh = self.chain_tip.clone();
+        self.commit_to(&bhh);
+    }
+    // This is used by miners, before renaming their committed blocks.
+    //   so that the block validation and processing logic doesn't
+    //   reprocess the same data as if it were already loaded
```

Comment:
So having read the code, let me see if I understand the flow here for metadata:

* When a miner is producing the block, their metadata `blockhash` values will start out as the sentinel block hash, and will ultimately be moved to the filename into which the MARF Trie gets written (per `commit_for_move()`)?
* When a miner and/or node is validating the block, their metadata `blockhash` values will start out as the sentinel block hash, and will ultimately be moved to the actual hash of the block (per `commit_to()`)?

Reply:
Yeah, that's correct.